src/WCSimAmBeGen.cc

```diff
@@ -0,0 +1,110 @@
+#include "WCSimAmBeGen.hh"
+#include "WCSimPrimaryGeneratorMessenger.hh"
+#include "WCSimPrimaryGeneratorAction.hh"
+
+#include "G4ParticleGun.hh"
+#include "G4PhysicalConstants.hh"
+#include "G4RandomDirection.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4SPSEneDistribution.hh"
+#include "G4SPSRandomGenerator.hh"
+#include "G4ParticleTable.hh"
+#include "G4Neutron.hh"
+#include "G4Gamma.hh"
+
+#include <random>
+
+using namespace std;
+
+WCSimAmBeGen::WCSimAmBeGen(){
+  // Initialise
+  this->Initialise();
+
+}
+
+WCSimAmBeGen::~WCSimAmBeGen(){
+  
+  // This needed to be deleted
+  delete myAmBeGun;
+  delete nEnergyDist;
+  delete rGen;
+}
+
+void WCSimAmBeGen::Initialise(){
+    nEnergyDist = new G4SPSEneDistribution();
+    rGen        = new G4SPSRandomGenerator();
+    myAmBeGun   = new G4ParticleGun();
+}
+
+G4double WCSimAmBeGen::GammaEnergy(){
+    
+    // Define the correspondent probabilities for every possible gamma scenario
+    probabilities = {0.26, 0.65, 0.08};
+    energies      = {0.0, 4.4, 7.7};
+
+    // Generate a weighted random number so we can select the gamma energy
+    mt19937 gen(std::random_device{}());
+    discrete_distribution<int> dist(probabilities.begin(), probabilities.end());
+    int index = dist(gen);
+    gEnergy = energies[index];
+    
+    return gEnergy; 
+}
+
+G4double WCSimAmBeGen::NeutronEnergy(G4double gEnergy){
+    // Depending on the gEnergy, we load the correspondent neutron energy spectrum
+    nEnergyDist->SetEnergyDisType("Arb");
+    
+    if (gEnergy == 0.0) {
+      nEnergyDist->ArbEnergyHistoFile("data/ground_state_spectrum.txt");
```

Comment:
personally I don't like using relative path as it may break easily

Reply:
you can use variable like [this](https://github.com/WCSim/WCSim/blob/abebcedc6216a942325484de65b1633d05b5b6fb/src/WCSimLIGen.cc#L36), then you don't even need to install those spectrum files

Reply:
Ok I reworked it to use a variable like in the LIGen. Thanks Ka Ming.